Allow UPDATE and DELETE for table variables inside functions

timchang514 opened the pull request:
### Description

Allow UPDATE and DELETE for Table Variables inside functions. Previously these actions were blocked since normal tables cannot be updated or deleted inside functions. However, table variables should be allowed to be targets of UPDATE/DELETE statements. 

### Issues Resolved
BABEL-2688

### Test Scenarios Covered ###
Unit tests included.


### Check List
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is under the terms of the Apache 2.0 and PostgreSQL licenses, and grant any person obtaining a copy of the contribution permission to relicense all or a portion of my contribution to the PostgreSQL License solely to contribute all or a portion of my contribution to the PostgreSQL open source project.

For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/babelfish-for-postgresql/babelfish_extensions/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

forestkeeper commented on test/JDBC/expected/BABEL-2688-vu-prepare.out:
```diff
@@ -0,0 +1,100 @@
+	UPDATE t
+	SET a = 1
+	FROM babel_2688_tab1 AS t;
+	RETURN 0;
```
I'm confused why this one has the error

timchang514 replied:
Shouldn't allow UPDATE on normal tables in functions, only table variables

timchang514 committed:
Allow UPDATE and DELETE for table variables in functions

Previously these actions were blocked since normal tables cannot be updated or deleted inside functions. However, table variables should be allowed to be targets of UPDATE/DELETE statements.

timchang514 commented:
failing test is due to github actions
